decode.go

```diff
@@ -610,6 +636,11 @@ func (d *Decoder) decodeStruct(dst reflect.Value, src ast.Node) error {
 			}
 		}
 	}
+	if len(unknownFields) != 0 && d.disallowUnknownField {
+		for key, node := range unknownFields {
+			return errors.ErrSyntax(fmt.Sprintf("unknown field \"%s\"", key), node.GetToken())
```

Comment:
There is no need to escape with double quotes if implemented in the following way :)

```go
fmt.Sprintf(`unknown field "%s"`,  key)
```